src/main/scala/it/pagopa/interop/partymanagement/api/impl/PartyApiServiceImpl.scala

```diff
@@ -121,6 +121,72 @@ class PartyApiServiceImpl(
 
   }
 
+  /** Code: 200, Message: successful operation, DataType: Institution
+   * Code: 405, Message: Institution not found, DataType: Problem
+   * Code: 400, Message: Invalid ID supplied, DataType: Problem
+   */
+  override def updateInstitutionById(id: String, institution: Institution)(implicit
+    toEntityMarshallerInstitution: ToEntityMarshaller[Institution],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    contexts: Seq[(String, String)]
+  ): Route = {
+    logger.info(s"Updating institution $id")
+
+    val commander: EntityRef[Command] = getCommander(id)
+
+    val updatedInstitution: Future[StatusReply[Party]] = for {
+      uuid             <- id.toFutureUUID
+      institutionParty <- commander.ask(ref => GetParty(uuid, ref)).flatMap {
+        case Some(x: InstitutionParty) => Future.successful(x)
+        case Some(x: PersonParty)      => Future.failed(InvalidParty("InstitutionParty", x.toString))
+        case None                      => Future.failed(UpdateInstitutionNotFound(id))
+      }
```

Comment:
no need to flatMap as we are already in a for comprehension in `Future`
```suggestion
      party <- commander.ask(ref => GetParty(uuid, ref))
      institutionParty <- party match {
        case Some(x: InstitutionParty) => Future.successful(x)
        case Some(x: PersonParty)      => Future.failed(InvalidParty("InstitutionParty", x.toString))
        case None                      => Future.failed(UpdateInstitutionNotFound(id))
      }
```